cpp/util/Histogram.h

```diff
@@ -22,20 +23,22 @@ namespace freud { namespace util {
  */
 template<typename T> struct Weight
 {
-    Weight() : value(1), is_default(true) {}
-    Weight(T value) : value(value), is_default(false) {}
+    Weight() : value(1) {}
```

Comment:
Is it possible to set the default weight below where the member is declared, like `T value {1};` or is that not possible because `T` is a template type parameter?

Reply:
Nope that should work, I just didn't notice it since it didn't get flagged. Of course, that would fail if you did something weird like `Weight<std::map<T, U>> x`... but I think we can safely assume nobody will ever use a non-numeric type `T`.

Reply:
Let's put in a template check or static assert for [is_arithmetic](https://en.cppreference.com/w/cpp/types/is_arithmetic).